drivers/serial/uart_mcux_lpuart.c

```diff
@@ -13,10 +13,16 @@
 #include <device.h>
 #include <drivers/uart.h>
 #include <drivers/clock_control.h>
+#ifdef CONFIG_PINCTRL
```

Comment:
Any reason this is conditionally included.

Reply:
This specific `ifdef` is not required, although it does reduce the total include dependencies when pinctrl is disabled. The remaining `ifdef`s in the driver are required, but do you think this one should be removed?

Reply:
Since we are transitioning to `PINCTRL` why should we provide an `non-PINCTRL` option?

Reply:
Long term, we should not. Short term, we need to to allow pinctrl to be conditional because boards that do not support it will fail to build using the driver if it requires pinctrl

Reply:
As long as you don't have all drivers for the given SoC converted from pinmux to pinctrl, you should keep #ifdefs in the drivers. Otherwise, end-users will end up using a hybrid pinmux/pinctrl solution, where _they_ need to investigate the source code of each enabled driver to see if they need to it up using pinctrl dts or pinmux.c.